cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,38 @@
-/// <reference types='cypress' />
-
 describe('Student Registration page', () => {
+  let user;
   before(() => {
 
+    cy.visit('/');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('', () => {
+  
+  it('should allow to register a user', () => {
+    cy.get('#firstName').type(user.firstName);
+    cy.get('#lastName').type(user.lastName);
+    cy.get('#userEmail').type(user.email);
+    cy.contains('.custom-radio', user.gender).click();
+    cy.get('#userNumber').type(user.number);
+    cy.get('#dateOfBirthInput').type('{selectAll}8 Jun 1998{enter}');
+    cy.get('.subjects-auto-complete__value-container').type(user.subject + "{enter}");
+    cy.contains('.custom-checkbox', user.hobby).click();
+    cy.get('#currentAddress').type(user.address);
+    cy.get('#state').type('{downarrow}{enter}');
+    cy.get('#city').type('{downarrow}{enter}');
+    cy.contains('#submit', 'Submit').click();
 
+    cy.get('#example-modal-sizes-title-lg').should('contain.text', 'Thanks for submitting the form');
+    cy.get('tbody > :nth-child(1) > :nth-child(2)').should('contain', user.firstName);
```

Comment:
Could you use the other selector, please? So we can understand what exactly row you are checking in the table.
Fix it for all further checks